experimental/adaptive-dialog/csharp_dotnetcore/calendar-bot-cui/Dialogs/AddContactDialog/AddContactDialog.cs

```diff
@@ -0,0 +1,306 @@
+                            new Foreach()
+                            {
+                                ListProperty = "dialog.AddContactDialog_UserAll.value",
+                                Steps = new List<IDialog>()
+                                {
+                                    new IfCondition()
+                                    {
+                                        Condition = "contains(dialog.AddContactDialog_UserAll.value[dialog.index].displayName, user.CreateCalendarEntry_PersonName) == true ||" +
+                                            "contains(dialog.AddContactDialog_UserAll.value[dialog.index].emailAddresses[0].address, user.CreateCalendarEntry_PersonName) == true",
+                                        Steps = new List<IDialog>()
+                                        {
+                                            new EditArray()
+                                            {
+                                                ArrayProperty = "dialog.matchedEmails",
+                                                ChangeType = EditArray.ArrayChangeType.Push,
+                                                Value = "dialog.AddContactDialog_UserAll.value[dialog.index].emailAddresses[0].address"
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    },
+                    new IfCondition()
+                    {
+                        Condition = "dialog.matchedEmails != null && count(dialog.matchedEmails) > 0",
+                        Steps = new List<IDialog>()
+                        {
+                            new SendActivity("[emailTemplate(dialog.matchedEmails[user.CreateCalendarEntry_pageIndex * 3], " +
+                                "dialog.matchedEmails[user.CreateCalendarEntry_pageIndex * 3 + 1]," +
+                                "dialog.matchedEmails[user.CreateCalendarEntry_pageIndex * 3 + 2])]"),// TODO only simple card right now, will use fancy card then\
+                            new DeleteProperty
+                            {
+                                Property = "turn.CreateCalendarEntry_Choice"
+                            },
+                            new ChoiceInput(){
+                                Property = "turn.CreateCalendarEntry_Choice",
+                                Prompt = new ActivityTemplate("[EnterYourChoice]"),
+                                Choices = new List<Choice>()
+                                {
+                                    new Choice("Confirm The First One"),
+                                    new Choice("Confirm The Second One"),
+                                    new Choice("Confirm The Third One")
+                                },
+                                Style = ListStyle.SuggestedAction
+                            },
```

Comment:
What if user's input is not one of the choices? Need to enable luis entity analysis here. (P2)